File: services/rust/actix/src/routes/middleware/log.rs
use actix_web::{
    Error,
    body::BoxBody,
    dev::{ServiceRequest, ServiceResponse},
    middleware::Next,
};
use tracing::{debug, instrument};

#[instrument(skip_all)]
pub async fn log_middleware(
    req: ServiceRequest,
    next: Next<BoxBody>,
) -> Result<ServiceResponse<BoxBody>, Error> {
    debug!(method = %req.method(), path = %req.uri(), "Incoming request");

    next.call(req).await
}

File: services/go/fiber/routes/middleware/log.go
package middleware

import (
	"log/slog"

	"github.com/gofiber/fiber/v3"
)

func LogMiddleware() fiber.Handler {
	return func(c fiber.Ctx) error {
		slog.Debug("Incoming request", "method", c.Method(), "path", c.Path())

		return c.Next()
	}
}
